frontend/src/common/hooks/useCategoryFilter/common/utils.ts

```diff
@@ -0,0 +1,114 @@
+import { Filters } from "react-table";
+import { removeOntologyTermIdPrefix } from "src/common/hooks/useCategoryFilter/common/multiPanelOntologyUtils";
+import { COLLATOR_CASE_INSENSITIVE } from "src/components/common/Filter/common/constants";
+import {
+  Categories,
+  CategoryFilter,
+  CategoryFilterConfig,
+  CategorySetValue,
+  CategoryValueId,
+  CATEGORY_FILTER_ID,
+  ETHNICITY_UNSPECIFIED_LABEL,
+  PUBLICATION_DATE_LABELS,
+  SelectCategoryValueView,
+} from "src/components/common/Filter/common/entities";
+
+/**
+ * General utils shared across the different category filter types.
+ */
+
+/**
+ * Find and return the selected values for the given category.
+ * @param categoryFilterId - ID of category to find selected filters of.
+ * @param filters - Current set of selected category values (values) keyed by category (id).
+ * @returns Array of filters
+ */
+export function getCategoryFilter<T extends Categories>(
+  categoryFilterId: CATEGORY_FILTER_ID,
+  filters: Filters<T>
+): CategoryFilter | undefined {
+  return filters.find((filter) => filter.id === categoryFilterId);
+}
+
+/**
+ * Build the display value for the given category and category value. For ontology terms, look up corresponding labels.
+ * @param config - Config model of category to build category value views for.
+ * @param categoryValueId - Category value to display (e.g. "normal").
+ * @param ontologyTermLabelsById - Set of ontology term labels keyed by term ID, used to determine labels for ontology
+ * @returns String to display as a label for the given category and category value.
+ */
+export function buildCategoryValueLabel(
+  config: CategoryFilterConfig,
+  categoryValueId: CategoryValueId,
+  ontologyTermLabelsById: Map<string, string>
```

Comment:
Added ontologyTermLabelsById to allow filtering of the new ontology fields to be driven by ontology term ID while retaining the label for display.